Assets/Scripts/Player/Health.cs

```diff
@@ -0,0 +1,251 @@
+            {
+                statusBools[target].bruised = true;
+            }
+
+            //Calculates if a limb is bleeding. If true, begin the BloodLoss Coroutine
+            if (statusBools[target].bleeding == false && Random.Range(Thresholds.thresholds[(int)Entities.Organic].minBleed, Thresholds.thresholds[(int)Entities.Organic].maxBleed) * limbHealthInfo[target].maxLimbHP < limbHealthInfo[target].brute)
+            {
+                statusBools[target].bleeding = true;
+                numBleedingLimbs++;
+                /*if(numBleedingLimbs == 1)
+                StartCoroutine("BloodLoss");
+                StartCoroutine("BloodRegen");
+                */
+            }
+
+            //Calculates if a limb is crippled
+            if (statusBools[target].crippled == false && Random.Range(Thresholds.thresholds[(int)Entities.Organic].minCrippled, Thresholds.thresholds[(int)Entities.Organic].maxCrippled) * limbHealthInfo[target].maxLimbHP < limbHealthInfo[target].brute)
+            {
+                statusBools[target].crippled = true;
+            }
+        }
+
+        //Determines the status of an affected limb based on the burn damage
+        else if (dmgType == 1)
+        {
+            //Calculates if a limb is burned
+            if (statusBools[target].burned == false && Random.Range(Thresholds.thresholds[(int)Entities.Organic].minBurn, Thresholds.thresholds[(int)Entities.Organic].maxBurn) * limbHealthInfo[target].maxLimbHP < limbHealthInfo[target].burn)
+            {
+                statusBools[target].burned = true;
+            }
+
+            //Calculates if a limb is numb
+            if (statusBools[target].numb == false && Random.Range(Thresholds.thresholds[(int)Entities.Organic].minNumb, Thresholds.thresholds[(int)Entities.Organic].maxNumb) * limbHealthInfo[target].maxLimbHP < limbHealthInfo[target].burn)
+            {
+                statusBools[target].numb = true;
+            }
+
+            //Calculates if a limb is blistered
+            if (statusBools[target].blistered == false && Random.Range(Thresholds.thresholds[(int)Entities.Organic].minBlistered, Thresholds.thresholds[(int)Entities.Organic].maxBlistered) * limbHealthInfo[target].maxLimbHP < limbHealthInfo[target].burn)
+            {
+                statusBools[target].blistered = true;
+            }
+
+        }
+
+        //Determines if the player is in soft crit or regular crit
+        if(scrit == false && currentHealth <= 0)
+        {
+            scrit = true;
+        }
+        if(crit == false && currentHealth <= -50)
+        {
+            scrit = false;
+            crit = true;
+        }
+    }
+
+    //Coroutine to lose blood overtime. Currently (10 * the number of bleeding limbs) cc's of blood are lost every 5 seconds
+    private IEnumerator BloodLoss()

```

Comment:
This function should not require a while loop within as long as you check requirements before calling it within whatever life loop you end up using.
i.e.
```
private void life()
{
    if(numBleedingLimbs)
    {
        handleBloodLoss()
    }
}

private IEnumerator handleBloodLoss()
{
    currentBloodVol = Max(currentBloodVol - (numBleedingLimbs * 10), 0)
}
````